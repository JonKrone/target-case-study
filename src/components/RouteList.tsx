import React from 'react'
import useFetch from 'use-http'
import { RouteItem } from './RouteItem'
import { useRouteMatch } from 'react-router-dom'
import { cardinalNumMap } from '../utils/utils'

export const RouteList: React.FC<{}> = () => {
  const match = useRouteMatch('/:route/:dir')
  const params: Record<string, string> = match?.params || {}

  // TODO(krone): handle error
  const { error, data = [] } = useFetch<NextTrip.Route[]>(
    { path: '/Routes', suspense: true },
    []
  )

  return (
    <nav
      style={{
        width: '50%',
        overflow: 'auto',
        borderBottomLeftRadius: '5px',
      }}
    >
      <ul
        style={{
          margin: 0,
          paddingLeft: 0,
          listStyle: 'none',
        }}
      >
        {data.map((route) => (
          <RouteItem
            key={route.Route}
            item={route}
            isActive={params.route === route.Route}
            activeDir={cardinalNumMap[params.dir]}
          />
        ))}
      </ul>
    </nav>
  )
}

export default RouteList
