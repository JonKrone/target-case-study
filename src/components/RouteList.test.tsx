import React from 'react'
import { render } from '@testing-library/react'
import useFetch from 'use-http'

import RouteList from './RouteList'

jest.mock('use-http', () => ({
  __esModule: true,
  default: jest.fn(() => mockUseFetchData),
}))

jest.mock('./RouteItem', () => ({
  __esModule: true,
  default: jest.fn(() => <>RouteItem</>),
}))

const mockUseFetchData = {
  data: [{ Description: 'METRO Blue', Route: '81', ProviderID: '1' }],
}

describe('RouteList', () => {
  it('renders', () => {
    const { asFragment } = render(<RouteList />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('invokes useFetch', () => {
    render(<RouteList />)
    expect(useFetch).toHaveBeenCalledWith(
      expect.objectContaining({ path: '/Directions/81' }),
      []
    )
  })
})
