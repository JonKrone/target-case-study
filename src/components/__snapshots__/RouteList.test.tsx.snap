// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteList renders 1`] = `
<DocumentFragment>
  <nav
    style="width: 50%; overflow: auto; border-bottom-left-radius: 5px;"
  >
    <ul
      style="margin: 0px; padding-left: 0px; list-style: none;"
    >
      RouteItem
    </ul>
  </nav>
</DocumentFragment>
`;
